gloo/common/linux.cc

```diff
@@ -198,7 +199,7 @@ static int getInterfaceSpeedGLinkSettings(int sock, struct ifreq* ifr) {
   } ecmd;
   int rv;
 
-  ifr->ifr_data = (__caddr_t)&ecmd;
+  ifr->ifr_data = (caddr_t)&ecmd;
```

Comment:
@stemann Why not `(void *)`?

Reply:
Okay, the Linux kernel defines this field as `char *`, probably for ergonomics ... https://lists.freebsd.org/pipermail/freebsd-questions/2006-August/128784.html

```
/opt/ai/pytorch/third_party/gloo/gloo/common/linux.cc:201:19: error: invalid conversion from 'void*' to 'char*' [-fpermissive]
  201 |   ifr->ifr_data = (void *)&ecmd;
      |                   ^~~~~~~~~~~~~
      |                   |
      |                   void*
/opt/ai/pytorch/third_party/gloo/gloo/common/linux.cc: In function 'int gloo::getInterfaceSpeedGSet(int, ifreq*)':
/opt/ai/pytorch/third_party/gloo/gloo/common/linux.cc:229:19: error: invalid conversion from 'void*' to 'char*' [-fpermissive]
  229 |   ifr->ifr_data = (void *)&edata;
      |                   ^~~~~~~~~~~~~~
      |                   |
      |                   void*
```

Reply:
Yeah, it seems the Linux kernel defines `caddr_t` as a typedef for `char *`.